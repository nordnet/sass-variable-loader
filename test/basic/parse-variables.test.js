const { parse } = require('../../src');
const units = require('./units.json');

const tests = {
  withoutComments({ sass }) {
    test('should contain "grayBase"', () => {
      const result = parse(sass);
      expect(typeof result).toBe('object');
      expect(result).toHaveProperty('grayBase');
      expect(result).toMatchSnapshot();
    });

    const options = { camelCase: false };
    const optionsStr = JSON.stringify(options);
    test(`should contain "gray-base" when options=${optionsStr}`, () => {
      const result = parse(sass, options);
      expect(typeof result).toBe('object');
      expect(result).toHaveProperty('gray-base');
      expect(result).toMatchSnapshot();
    });
  },

  withComments({ sass }) {
    const result = parse(sass);

    test('should return an object with a key "one"', () => {
      expect(typeof result).toBe('object');
      expect(result).toHaveProperty('one');
      expect(result).toMatchSnapshot();
    });

    test('should return an object without a key "y"', () => {
      expect(result).not.toHaveProperty('y');
    });
  },

  indentedSass({ sass }) {
    const result = parse(sass, { indented: true });

    test('should return an object with a key "one"', () => {
      expect(typeof result).toBe('object');
      expect(result).toHaveProperty('one');
      expect(result).toMatchSnapshot();
    });
  },

  emptySassFile({ sass }) {
    const result = parse(sass);

    test('should be an empty object', () => {
      expect(result).toEqual({});
    });
  },
};

describe('parseVariables()', () => {
  units.forEach(unit => {
    const testFn = tests[unit.name];
    if (!testFn) {
      return;
    }
    describe(unit.title, () => {
      testFn(unit);
    });
  });
});
