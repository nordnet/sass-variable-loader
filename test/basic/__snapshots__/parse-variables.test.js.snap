// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseVariables() indented sass should return an object with a key "one" 1`] = `
Object {
  "one": "123",
  "x": "123",
}
`;

exports[`parseVariables() with comments should return an object with a key "one" 1`] = `
Object {
  "one": "123",
  "x": "123",
}
`;

exports[`parseVariables() without comments should contain "gray-base" when options={"camelCase":false} 1`] = `
Object {
  "gray": "#555555",
  "gray-base": "#000",
  "gray-dark": "#333333",
  "gray-darker": "#222222",
  "gray-light": "#777777",
  "gray-lighter": "#eeeeee",
}
`;

exports[`parseVariables() without comments should contain "grayBase" 1`] = `
Object {
  "gray": "#555555",
  "grayBase": "#000",
  "grayDark": "#333333",
  "grayDarker": "#222222",
  "grayLight": "#777777",
  "grayLighter": "#eeeeee",
}
`;
